services/github/github-commit-activity.service.js

```diff
@@ -115,6 +152,20 @@ export default class GitHubCommitActivity extends GithubAuthV4Service {
     return repo.history.totalCount
   }
 
+  static transformAuthorFilter({ res, buffer }) {
+    if (buffer.message === 'Not Found') {
+      throw new InvalidResponse({ prettyMessage: 'invalid branch' })
+    }
```

Comment:
Do we ever hit this line?
If I call something like  `/github/commit-activity/t/badges/shields/does-not-exist?authorFilter=chris48s` it tells me "repo not found". If we can't distinguish between branch not found and repo not found then we can just make the other error "repo or branch not found" (this is quite common).

While I was testing this, I also noticed that the branch not found case doesn't seem to be hanled correctly. It is not new in this PR (already exists in prod https://shields.io/github/commit-activity/t/badges/shields/does-not-exist ), but if you have a chance to look at it while you are looking at this, that would be amazing.

Given neither of these work but the spec tests pass, I'd suggest we move the "branch not found" test cases out to the service tests layer.

Reply:
Hmm. I am not following. What do you mean when you say

> You are correct as well about the `interval='t'` not being used

?

This behaves the same way if I call https://shields.io/github/commit-activity/m/badges/shields/does-not-exist

Reply:
My bad, there is no diffrence, no idea how i mixed this in my reply. I edited the reply as i was debugging.
Should have read it twice before posting.

Reply:
Could we add a couple of test cases to the service tests file (`.tester.js`) covering invalid branch (for both the GraphQL and Rest paths).

Reply:
I separated the GraphQL test and fixed the error handling in PR #9258 as this is not related directly to this change, i think its better for history.
I will add another test for the REST path in this PR